app/enquiries/common/datahub_utils.py

```
@@ -39,11 +42,14 @@ def dh_request(method, url, payload, request_headers=None, timeout=15):
     if request_headers:
         headers = request_headers
     else:
-        # TODO: We don't need to send the access token in the headers
-        # once SSO is integrated as it comes from SSO directly
+        # Extract access token
+        if not access_token:
+            session = get_oauth_payload(request)
+            access_token = session["access_token"] if session else 'invalid-token'
```

Wondering if you can fail early when `invalid-token` rather than attempting and failing eventually.

---

I will look into it but in case of testing it will be invalid as this is going to be mocked.

---

To bail out early at this stage I need to construct response object, so I think it is better to continue and fail at next step where we get response object with error description etc so will leave it as is.

---

Don't think we should be deliberately making a request with 'invalid-token'. When would the access token be missing?

---

checked again, this check is not required. It was only required for tests but we are mocking the response so it is not required.